resources-domain/resources-service/src/admin/adminSearchDataAccess.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import { AccountSID, FlatResource } from '@tech-matters/types';
+import { db } from '../connection-pool';
+import { generateSelectResourcesForReindexSql } from './sql/adminSearchSql';
+
+export const getResourcesBatchForReindexing = async ({
+  lastUpdatedFrom,
+  lastUpdatedTo,
+  accountSid,
+  resourceIds,
+  start = 0,
+  limit = 1000,
+}: {
+  lastUpdatedFrom?: string;
+  lastUpdatedTo?: string;
+  accountSid?: AccountSID;
+  resourceIds?: string[];
+  start?: number;
+  limit?: number;
+}): Promise<FlatResource[]> => {
+  return db.task(async tx => {
+    return tx.manyOrNone(generateSelectResourcesForReindexSql(Boolean(resourceIds)), {
+      lastUpdatedFrom,
+      lastUpdatedTo,
+      accountSid,
+      resourceIds,
+      start,
+      limit,
```

Comment:
I know the description says is intentional but can this having no upper limit affect the DB?

Reply:
NVM, I get it now, https://github.com/techmatters/hrm/pull/375 solves it [with streams](https://www.npmjs.com/package/pg-query-stream#:~:text=Using%20manual%20limit%20and%20offset%20queries%20to%20fake%20out%20async%20itteration%20through%20your%20data%20is%20cumbersome%2C%20and%20way%20way%20way%20slower%20than%20using%20a%20cursor.).

Reply:
In this implementation it determines the maximum number of records in each batch we pull back from the DB, but does not limit the overall number of resources sent for reindexing. The default of 1000 is probably fine for prod, but it gets overridden with a low value for the service tests - just so the batching logic 'kicks in' with the small test dataset and gets tested